math/vecops/inc/ROOT/RVec.hxx

```diff
@@ -3242,6 +3242,274 @@ RVec<typename RVec<T>::size_type> Enumerate(const RVec<T> &v)
+ * \par Example code, at the ROOT prompt:
+ * ~~~{.cpp}
+ * using namespace ROOT::VecOps;
+ * cout << Logspace(4, 10, 12) << '\n';
+ * // { 10000, 35111.9, 123285, 432876, 1.51991e+06, 5.3367e+06, 1.87382e+07, 6.57933e+07, 2.31013e+08, 8.11131e+08, 2.84804e+09, 1e+10 }
+ * cout << Logspace(0, 0, 50) << '\n';
+ * // { 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 }
+ * cout << Logspace(0, 0, 0) << '\n';
+ * // {  }
+ * cout << Logspace(4, 10, 12, 10.0, false) << '\n';
+ * // { 10000, 31622.8, 100000, 316228, 1e+06, 3.16228e+06, 1e+07, 3.16228e+07, 1e+08, 3.16228e+08, 1e+09, 3.16228e+09 }
+ * cout << Logspace<int, int>(1, 5, 3) << '\n';
+ * // { 10, 1000, 100000 }
+ * ~~~
+ */
+template <typename T = double, typename Ret_t = std::conditional_t<std::is_floating_point_v<T>, T, double>>
+inline RVec<Ret_t> Logspace(T start, T end, unsigned long long n = 128, const bool endpoint = true, T base = 10.0)
+{
+    if (!n || (n > std::numeric_limits<long long>::max())) // Check for invalid or absurd n.
+    {
+        return {};
+    }
+    RVec<Ret_t> temp(n);
+    
+    long double start_c = start;
+    long double end_c   = end;
+    long double base_c  = base;
+    
+    long double step = (end_c - start_c) / (n - endpoint);
+    
+    temp[0] = std::is_floating_point_v<Ret_t> ?
+    static_cast<Ret_t>(std::pow(base_c, start_c)) :
+    std::floor(std::pow(base_c, start_c));
+     
+    if (std::is_floating_point_v<Ret_t>)
+    {
+        for (unsigned long long i = 1; i < n; i++)
+        {
+            auto exponent = start_c + i * step;
+            temp[i] = static_cast<Ret_t>(std::pow(base_c, exponent));
+        }
+    }
+    else
+    {
+        for (unsigned long long i = 1; i < n; i++)
+        {
+            auto exponent = start_c + i * step;
+            temp[i] = std::floor(std::pow(base_c, exponent));
+        }
+    }
+     
+    return temp;
+}
+
+/**
+ * \brief Produce RVec with entries in the range [start, end) in increments of step.
+ *
+ * This function generates a vector of values starting at \p start and incremented by \p step,
+ * continuing until the values reach or exceed \p end (the interval is half-open: [start, end)).
+ * The number of elements is computed as:
+ * \f[
+ * n = \lceil \frac{\text{end} - \text{start}}{\text{step}} \rceil
+ * \f]
+ * ensuring that the arithmetic is performed in a floating-point context when needed.
+ *
+ * The function is templated to allow for different arithmetic types. The deduced type \c Ret_t, if not
+ * explicitly specified, is determined as follows: if \p T is a floating point type, that type is used;
+ * otherwise, the arithmetic is performed using \c double.
+ *
+ * \tparam T Type of the start, end, and step values. Default is double.
+ * \tparam Ret_t Deduced type used for arithmetic, which, if not explicitly
+ * specified, is \p T if that is a floating point type, or double otherwise.
```

Comment:
```suggestion
 * The function is templated to allow for different return types. The return type \c Ret_t, if not
 * explicitly specified, is determined as follows: if \p T is a floating point type, that type is used;
 * otherwise, the return type is \c double.
 *
 * \tparam T Type of the start, end, and step values. Default is double.
 * \tparam Ret_t Return type, which, if not explicitly
 * specified, is \p T if that is a floating point type, or double otherwise.
```